fecgov/fec-cms
# Upgrade npm to latest

## Summary (required)

Upgrade NPM to latest  version 7.24.0

- Resolves #4591

(Include a summary of proposed changes and connect issue below)
- update NPM version in package.json
- generate package-local.json with latest npm changes

### Required reviewers

At least two developers required


## Screenshots

**Before NPM package upgrade**:

```
(cms3710) macadmins-mbp-5:fec-cms pkasireddy$ node -v
v10.16.0
(cms3710) macadmins-mbp-5:fec-cms pkasireddy$ npm -v
6.14.3
```

**After NPM package upgrade**:
```
(cms3710) macadmins-mbp-5:fec pkasireddy$ node -v
v14.13.0
(cms3710) macadmins-mbp-5:fec pkasireddy$ npm -v
7.24.0
```

## How to test

- checkout branch
- create new python virtual env: run `pyenv virtualenv 3.7.10 cms3710` 
- install requirements: run `pip install -r requirements.txt`
- check the local  npm version before upgrade:
    - run `npm -v` : terminal  display `6.14.*`
- remove existing node_modules folder on fec-cms repo: run `rm -rf node-modules`
- clear npm cache: run `npm cache clean -f`
- install/update to latest npm:  run `npm install -g npm@7.24.0`
- verify the npm version: run `npm -v` : terminal display `7.24.0`
- run `python manage.py migrate`, just in case
- run `npm i`
- run `npm run build`
- run `npm run build-sass`, just in case 
- run `npm run build-js`, just in case
- run `npm run test-single`, just in case
- start server and test website  (http://127.0.0.1:8000/): run `./manage.py runserver`
- make sure no errors on browser or the terminal logs
